src/diff.c

```diff
@@ -276,10 +276,20 @@ static int64_t search(int64_t *I, u_char *old, int64_t oldsize,
 	}
 
 	x = st + (en - st) / 2;
-	if (memcmp(old + I[x], new, MIN(oldsize - I[x], newsize)) < 0) {
+
+	int64_t length = MIN(oldsize - I[x], newsize);
+	int result = memcmp(old + I[x], new, length);
+
+	if (result < 0) {
 		return search(I, old, oldsize, new, newsize, x, en, pos);
-	} else {
+	} else if (result > 0) {
 		return search(I, old, oldsize, new, newsize, st, x, pos);
+	} else {
```

Comment:
Technically, yes, but this is a style preference. I think having the `else` block keeps the code in a more maintainable state, since it is linked (conceptually) to the `memcmp` result.